OADP-3261: OADP 1.3 Performance & Scale (#69704)
openshift/openshift-docs: modules/oadp-backup-restore-for-large-usage.adoc

// Module included in the following assemblies:
// about-installing-oadp.adoc

:_mod-docs-content-type: CONCEPT
[id="oadp-backup-restore-for-large-usage_{context}"]
= NodeAgent CPU for large usage

> Testing shows that increasing `NodeAgent` CPU can significantly improve backup and restore times when using {oadp}.

Reviewer: ```suggestion
Testing shows that increasing `NodeAgent` CPU can significantly improve backup and restore times when using {oadp-short}.
```

Author: @kaovilai Since it is the first usage in the module, it actually should be {oadp-first}, but thanks for keeping me honest. :)

It is not recommended to use Kopia without limits in production environments on nodes running production workloads due to Kopia’s aggressive consumption of resources.

However, running Kopia with limits that are too low results in CPU limiting and slow-performing backups and restore situations.

> The testing undertaken observed that running Kopia with limits of around 20 cores and 32 Gi memory covered most of our largest cases effectively.

Reviewer: `our largest cases`

I would like to see a specific number quoted here. Some cu might have real world cases exceeding `our largest cases` which could invalidate some of the suggestions here.

// Module included in the following assemblies:

> // about-installing-oadp.adoc

Reviewer 2: ```suggestion
// * backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc
```

:_mod-docs-content-type: CONCEPT
[id="oadp-backup-restore-for-large-usage_{context}"]
= NodeAgent CPU for large usage

> Testing shows that increasing `NodeAgent` CPU can significantly improve backup and restore times when using {oadp-first}.

Reviewer 2: Up to you!
```suggestion
Increasing `NodeAgent` CPU can significantly improve backup and restore times when using {oadp-first}.
```

Author: Prefer to leave as is.

> It is not recommended to use Kopia without limits in production environments on nodes running production workloads due to Kopia’s aggressive consumption of resources.

Reviewer 2: Maybe this could be an IMPORTANT block, but up to you!

> 

Reviewer 2: Wherever you end up putting it, I do think these sentences should be in the same paragraph and not split between lines 
```suggestion
```

> However, running Kopia with limits that are too low results in CPU limiting and slow-performing backups and restore situations.

Reviewer 2: I think just `slow` is more fluid but this is up to you!
`slow-performing` -> `slow`

> The testing undertaken observed that running Kopia with 20 cores and 32 Gi memory supportedbackup and restore operations of over 100GB of data, multiple namespaces, or over 2000 pods in a single namespace.

Reviewer 3: NIT: missing space between `supported` and `backup`

Reviewer: +1

> The testing undertaken observed that running Kopia with 20 cores and 32 Gi memory supported backup and restore operations of over 100GB of data, multiple namespaces, or over 2000 pods in a single namespace.

Reviewer 2: I don't think there needs to be a space between the <num> and <gi>, but I could be wrong
```suggestion
The testing observed that running Kopia with 20 cores and 32Gi memory supported backup and restore operations of over 100GB of data, multiple namespaces, or over 2000 pods in a single namespace.
```

> Testing detected no CPU limiting or memory saturation with these resource specifications.

Reviewer 2: I also think this can be included with the previous sentence as one paragraph.

You can set these limits in Ceph MDS pods by following the procedure in https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.14/html/troubleshooting_openshift_data_foundation/changing-resources-for-the-openshift-data-foundation-components_rhodf#changing_the_cpu_and_memory_resources_on_the_rook_ceph_pods[Changing the CPU and memory resources on the rook-ceph pods].

> You need to add the following lines to the storage cluster Custom Resource (CR):

Reviewer 2: Maybe say what this config does like: 

`You need to add the following lines to the storage cluster Custom Resource (CR) <to add blank blank or what adding these parameters does> `
